sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubProducerClient.cs

```diff
@@ -59,88 +67,257 @@ public class EventHubProducer : IAsyncDisposable
+            Argument.AssertNotNullOrEmpty(connectionString, nameof(connectionString));
+            producerOptions = producerOptions?.Clone() ?? new EventHubProducerClientOptions();
+
+            OwnsConnection = true;
+            Connection = new EventHubConnection(connectionString, eventHubName, producerOptions.ConnectionOptions);
+            PartitionId = producerOptions.PartitionId;
+            RetryPolicy = producerOptions.RetryOptions.ToRetryPolicy();
+            InnerProducer = Connection.CreateTransportProducer(producerOptions);
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="EventHubProducerClient"/> class.
+        /// </summary>
+        ///
+        /// <param name="fullyQualifiedNamespace">The fully qualified Event Hubs namespace to connect to.  This is likely to be similar to <c>{yournamespace}.servicebus.windows.net</c>.</param>
+        /// <param name="eventHubName">The name of the specific Event Hub to associated the connection with.</param>
+        /// <param name="credential">The Azure managed identity credential to use for authorization.  Access controls may be specified by the Event Hubs namespace or the requested Event Hub, depending on Azure configuration.</param>
+        /// <param name="producerOptions">A set of options to apply when configuring the producer.</param>
+        ///
+        public EventHubProducerClient(string fullyQualifiedNamespace,
+                                      string eventHubName,
+                                      TokenCredential credential,
+                                      EventHubProducerClientOptions producerOptions = default)
+        {
+            Argument.AssertNotNullOrEmpty(fullyQualifiedNamespace, nameof(fullyQualifiedNamespace));
             Argument.AssertNotNullOrEmpty(eventHubName, nameof(eventHubName));
-            Argument.AssertNotNull(producerOptions, nameof(producerOptions));
-            Argument.AssertNotNull(retryPolicy, nameof(retryPolicy));
+            Argument.AssertNotNull(credential, nameof(credential));
+
+            producerOptions = producerOptions?.Clone() ?? new EventHubProducerClientOptions();
 
+            OwnsConnection = true;
+            Connection = new EventHubConnection(fullyQualifiedNamespace, eventHubName, credential, producerOptions.ConnectionOptions);
             PartitionId = producerOptions.PartitionId;
-            EventHubName = eventHubName;
-            Options = producerOptions;
+            RetryPolicy = producerOptions.RetryOptions.ToRetryPolicy();
+            InnerProducer = Connection.CreateTransportProducer(producerOptions);
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="EventHubProducerClient"/> class.
+        /// </summary>
+        ///
+        /// <param name="connection">The <see cref="EventHubConnection" />connection to use for communication with the Event Hubs service.</param>
```

Comment:
nit: missing blank space between `/>` and `connection`. Like in my previous comment, searching for `/>connection` might be a good idea.

Reply:
Got those with a global as well; lots of copy+paste there.